config/src/main/scala/bloop/config/Config.scala

```diff
@@ -59,6 +59,46 @@ object Config {
     private[bloop] val empty: Scala = Scala("", "", "", Array(), Array())
   }
 
+  sealed abstract class Platform(val name: String)
+  object Platform {
+    private[bloop] val default: Platform = JVM
+
+    case object JS extends Platform("js")
+    case object JVM extends Platform("jvm")
+    case object Native extends Platform("native")
+
+    def apply(platform: String): Platform = platform.toLowerCase match {
+      case JS.name => JS
+      case JVM.name => JVM
+      case Native.name => Native
+      case _ => throw new IllegalArgumentException(s"Unknown platform: '$platform'")
+    }
+  }
+
+  case class NativeConfig(
```

Comment:
It would be great if there was a scaladoc explaining each of the fields of this case class, as well as the options that they can take (e.g. `gc`). If this is too cumbersome, a link to some kind of Scala Native documentation would also be helpful